src/pages/Admin/Charity/Banking/CurrencySelector.tsx

```diff
@@ -0,0 +1,97 @@
+import { Combobox, Transition } from "@headlessui/react";
+import { Fragment, useState } from "react";
+import { WiseCurrency } from "types/aws";
+import { DrawerIcon } from "components/Icon";
+import { isEmpty } from "helpers";
+
+export type Currency = {
+  code: string;
+  name: string;
+  searchKeywords: string[];
+};
+
+type Props = {
+  value: Currency;
+  currencies: Currency[];
+  onChange: (currency: Currency) => void;
+};
+
+export default function CurrencySelector(props: Props) {
+  const [query, setQuery] = useState("");
+
+  const filteredCurrencies =
+    query === ""
+      ? props.currencies
+      : props.currencies.filter((currency) => {
+          // check whether query matches either the currency name or any of its keywords
+          const formatQuery = query.toLowerCase().replace(/\s+/g, ""); // ignore spaces and casing
+          const matchesName = currency.name
+            .toLowerCase()
+            .replace(/\s+/g, "") // ignore spaces and casing
+            .includes(formatQuery);
+          const matchesKeyword = new RegExp(
+            currency.searchKeywords.join("|"),
```

Comment:
seems like `searhKeyWords` is not a good filter attribute 
examples
search for `JPY`, or `PHP`, or `AUD`, and results are still broad

I tried `matchCode || matchName` and it yields more specific filter result

Reply:
> search for `JPY`, or `PHP`, or `AUD`, and results are still broad

I tested these values out with the current setup (available currencies Japanese yen and Australian dollar):
- searching by `JPY` returns only "Japanese yen" 
- searching by `PHP` returns nothing
- searching by `AUD` returns only "Australian dollar" 

What problem did you experience?

#### Explanation of the purpose of `searchKeywords`

Field `searchKeywords` is created by assigning [[wise currency].countryKeywords](https://docs.wise.com/api-docs/api-reference/currencies#object), the purpose of which is to aid the user in finding the desired currency.
You can check this by using Wise's URL to get all currencies:
https://api.sandbox.transferwise.tech/v1/currencies

The result looks like:
```js
[
    // ...
    {
        "code": "IDR",
        "symbol": "Rp",
        "name": "Indonesian rupiah",
        "countryKeywords": ["idn", "IDR", "ID", "Indonesia"],
        "supportsDecimals": false
    },
    {
        "code": "ILS",
        "symbol": "₪",
        "name": "Israeli shekel",
        "countryKeywords": [
            "ILS",
            "IL",
            "PS",
            "pse",
            "isr",
            "Israel",
            "Palestine"
        ],
        "supportsDecimals": true
    },
    // ...
]
```
Will add an explanatory comment to the code.

Reply:
@ap-justin did my explanation alleviate your concerns?

Reply:
> Field searchKeywords is created by assigning [[wise currency].countryKeywords](https://docs.wise.com/api-docs/api-reference/currencies#object), the purpose of which is to aid the user in finding the desired currency.
You can check this by using Wise's URL to get all currencies:
https://api.sandbox.transferwise.tech/v1/currencies

currently searching experience based on this is not good 
<img width="578" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/7c9d8b01-6df4-4530-8c87-5d38a9eabfd2">


<img width="560" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/7e88b766-adc1-4afb-83bd-510d7ffea225">


if you just based it on `code` and `name`, these are the results
<img width="545" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/db80285a-28bd-44a9-8c25-2ed04586a97a">

<img width="567" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/dec8891c-b798-48ed-9ecf-51f3085c3835">

<img width="583" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/3f870870-c9e0-4539-86f2-0f7c49569970">


it just doesn't make sense for people to use these keywords to search for their currency 
```
 {
    "code": "EUR",
    "symbol": "€",
    "name": "Euro",
    "countryKeywords": [
      "Cambodia",
      "cck",
      "fin",
      "Paraguay",
      "lux",
      "uga",
      "iot",
      "Solomon Islands",
      "Montserrat",
      "deu",
      "lva",
      "egy",
      "Guadeloupe",
      "hnd",
      "nzl",
      "AD",
      "fji",
      "Falkland Islands",
      "AE",
      "AG",
      "Seychelles",
```

Reply:
Found the main culprit for such a long list of (unintuitive) matches - it was a bug in the filter function. 

Nevertheless, I see what you mean. Even with a shorter list of matches, some of the keywords used by Wise just makes for nonsensical result list. This is because Wise's logic goes like "when a user searches for a currency code in a specific country, we will match not only the currency itself, but all the currencies that are available in said country". This is why whatever you write in the search box, USD and EURO will almost certainly pop up - because they are possible bank account currencies in most of the countries of the world.

Updated currency filter to always match only by code or name (ignore `WiseCurrency.countryKeywords`), see commit https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2439/commits/c4b02326a82a0ba562b1af80a756ce0e7d6e4e57